File: 03-interface/backend/app/schemas/group.py
"""
Schémas Pydantic pour le modèle Group.
"""
from pydantic import BaseModel
from typing import Optional, List
from enum import Enum


class GroupType(str, Enum):
    """Types de groupes."""
    DEPARTMENT = "department"
    PROJECT = "project"
    RECURRING = "recurring"


# ============================================================
# Schémas de base
# ============================================================

class GroupBase(BaseModel):
    name: str
    description: Optional[str] = None
    type: GroupType = GroupType.DEPARTMENT


class GroupCreate(GroupBase):
    """Schéma pour la création d'un groupe."""
    pass


class GroupUpdate(BaseModel):
    """Schéma pour la mise à jour d'un groupe."""
    name: Optional[str] = None
    description: Optional[str] = None
    type: Optional[GroupType] = None
    is_active: Optional[bool] = None


# ============================================================
# Schémas de réponse
# ============================================================

class GroupRead(GroupBase):
    """Réponse complète d'un groupe."""
    id: int
    is_active: bool

    class Config:
        from_attributes = True


class GroupWithCount(GroupRead):
    """Groupe avec compteur de membres pour les vues admin."""
    member_count: int = 0


class GroupMinimal(BaseModel):
    """Info groupe minimale pour le contexte utilisateur/meeting."""
    id: int
    name: str
    type: GroupType

    class Config:
        from_attributes = True

File: 03-interface/frontend-nextjs/src/types/group.ts
// Group Types
export type GroupType = "department" | "project" | "recurring";

export interface Group {
    id: number;
    name: string;
    type: GroupType;
    description?: string;
    created_at: string;
    updated_at: string;
}

export interface GroupCreate {
    name: string;
    type: GroupType;
    description?: string;
}

export interface GroupUpdate {
    name?: string;
    type?: GroupType;
    description?: string;
}

// Group type display helpers
export const groupTypeLabels: Record<GroupType, string> = {
    department: "Département",
    project: "Projet",
    recurring: "Récurrent",
};

export const groupTypeIcons: Record<GroupType, string> = {
    department: "🏢",
    project: "📁",
    recurring: "📅",
};
